src/python/visclaw/data.py

```diff
@@ -25,7 +25,7 @@ class ClawPlotData(clawdata.ClawData):
     """
 
     # ========== Initialization routine ======================================
-    def __init__(self, controller=None):
+    def __init__(self, controller=None, num_dim=2):
```

Comment:
True.  Seems to me the best way to approach this is your suggestion to move num_dim as an argument to getgauge, with a default value of 2.  Granted, this still won't be correct for non-2d runs... but at least it will only be wrong in getgauge (and GaugeData has it's own num_dim attribute that has a default value of 2d anyways).